js/src/components/PrimaryTaxonomyPicker.js

```diff
@@ -0,0 +1,262 @@
+	 *
+	 * @returns {Array} The selected terms.
+	 */
+	getSelectedTerms( terms, selectedTermIds ) {
+		return terms.filter( term => {
+			return selectedTermIds.includes( term.id );
+		} );
+	}
+
+	/**
+	 * Updates the state with the selected terms.
+	 *
+	 * @param {Array} terms           The available terms.
+	 * @param {Array} selectedTermIds The ids of the selected terms.
+	 *
+	 * @returns {void}
+	 */
+	updateSelectedTerms( terms, selectedTermIds ) {
+		const selectedTerms = this.getSelectedTerms( terms, selectedTermIds );
+		this.setState( {
+			selectedTerms,
+		} );
+	}
+
+	/**
+	 * Handles an onChange event.
+	 *
+	 * Updates the primary taxonomy in the store, as well as the replacement variable and the hidden field.
+	 *
+	 * @param {number} termId The term's id.
+	 *
+	 * @returns {void}
+	 */
+	onChange( termId ) {
+		const termIdNum = typeof termId === "string" ? parseInt( termId, 10 ) : termId;
+
+		const { name } = this.props.taxonomy;
+
+		this.updateReplacementVariable( termIdNum );
+
+		this.props.setPrimaryTaxonomyId( name, termIdNum );
+
+		this.input.value = termIdNum === -1 ? "" : termIdNum;
+	}
+
+	/**
+	 * Updates the primary taxonomy replacement variable.
+	 *
+	 * @param {number} termId The term's id.
+	 *
+	 * @returns {void}
+	 */
+	updateReplacementVariable( termId ) {
+		/**
+		 * We only use the primary category replacement variable, therefore only do this for the
+		 * category taxonomy.
+		 */
+		if ( this.props.taxonomy.name !== "category" ) {
+			return;
+		}
+		const primaryTerm = this.state.selectedTerms.find( term => term.id === termId );
+		if ( primaryTerm ) {
+			this.props.updateReplacementVariable( `primary_${ this.props.taxonomy.name }`, primaryTerm.name );
+		}
+	}
+
+	/**
+	 * Renders the PrimaryTaxonomyPicker component.
+	 *
+	 * @returns {ReactElement} The rendered component.
+	 */
+	render() {
+		const {
+			primaryTaxonomy,
+			taxonomy,
+		} = this.props;
+
+		const fieldId = `yoast-primary-${ taxonomy.name }-picker`;
+
+		return (
+			<div className="components-base-control__field">
+				<PrimaryTaxonomyPickerLabel
+					htmlFor={ fieldId }
+					className="components-base-control__label">
+					{
+						sprintf(
+							/* Translators: %s expands to the taxonomy name */
+							__( "Select the primary %s" ),
+							taxonomy.singular_label.toLowerCase()
+						)
+					}
+				</PrimaryTaxonomyPickerLabel>
+				<TaxonomyPicker
+					value={ primaryTaxonomy }
+					onChange={ this.onChange }
+					id={ fieldId }
+					terms={ this.state.selectedTerms }/>
+			</div>
+		);
+	}
+}
+
+PrimaryTaxonomyPicker.propTypes = {
+	selectedTermIds: PropTypes.arrayOf( PropTypes.number ),
+	primaryTaxonomy: PropTypes.string,
+	setPrimaryTaxonomyId: PropTypes.func,
+	updateReplacementVariable: PropTypes.func,
+	receiveEntityRecords: PropTypes.func,
+	taxonomy: PropTypes.shape( {
+		name: PropTypes.string,
+		/* eslint-disable */
+		field_id: PropTypes.string,
+		rest_base: PropTypes.string,
```

Comment:
You are introducing the `rest_base` in this pull. I would suggest to use `restBase` instead. But it is no big deal if you do not change it.